### ISSUE TITLE
`named_args` should support custom error types

### ISSUE THREAD

**Issue by TyOverby** [rust-bakery/nom#455]

**Comment by link2xt** [comment]
Are `named` and `named_args` sort of deprecated now? Front page of documentations says "For new parsers, we recommend using the functions instead of macros, since rustc messages will be much easier to understand.", this issue is not moved to milestone 5, and syntax for custom errors is not going to be easy to read because of backwards compatibility.

**Comment by Geal** [comment]
yup, deprecated for a while now. The function based parsers made it way easier
